torch/nn/parameter.py

```diff
@@ -98,35 +84,35 @@ def materialize(self, shape, device=None, dtype=None):
         if dtype is None:
             dtype = self.data.dtype
         self.data = torch.empty(shape, device=device, dtype=dtype)
-        self.__class__ = Parameter
+        self.__class__ = self.cls_to_become
 
     @property
     def shape(self):
         raise RuntimeError(
-            'Can\'t access the shape of an uninitialized parameter. '
+            'Can\'t access the shape of an uninitialized parameter or buffer. '
             'This error usually happens in `load_state_dict` when trying to load '
             'an uninitialized parameter into an initialized one. '
             'Call `forward` to initialize the parameters before accessing their attributes.')
 
     def share_memory_(self):
         raise RuntimeError(
-            'Can\'t share memory on an uninitialized parameter. '
+            'Can\'t share memory on an uninitialized parameter or buffer. '
             'Call `forward` to initialize the parameters before calling '
             '`module.share_memory()`.')
 
     def __repr__(self):
-        return 'Uninitialized parameter'
+        return '<' + self.__class__.__name__ + '>'
```

Comment:
nit: now you can use f string to make these cleaner: `f'<{self.__class__.__name__}>'`!